refactor!: make bearer a user concern

oetyng opened the pull request:
This removes the whole `Owner/OwnerOnce` recursive and confusing relationship, which infused the "bearer" concept into this Dbc library, and unnecessarily complicated the code.

Instead we use a simpler terminology, where we talk of the (`MainKey`:`PublicAddress`) pair, and the  (`DerivedKey`:`DbcId`) pair.

The api is now much simpler, as the possible concept of a "bearer" is a concern of the user of this library.

oetyng commented on src/dbc_id.rs:
```diff
@@ -0,0 +1,269 @@
+/// The secret key source has a static public address, which
+/// is shared with others in order to receive payments.
+/// With this key source, new secret keys can be generated.
+pub struct MasterKey(SerdeSecret<SecretKey>);
```
Oh, wow. Hehe, you're right. Didn't occur to me.
I guess MainKey is quite good actually 👍

davidrusu reviewed (dismissed):
Generally looks good, I'd just rename the MasterKey if anything.

oetyng committed:
refactor!: make bearer a user concern
- This removes the whole Owner/OwnerOnce recursive and confusing
relationship, and replaces it with a simpler terminology, where we talk
of master key, public address, dbc key and dbc id.
